collated/main/Jemereny.md

```diff
@@ -0,0 +1,995 @@
+            "There should be a valid location to the picture, the picture must be a .png ";
+    public static final String MESSAGE_PROFILEPICTURE_ERROR =
+            "Error copying file.";
+
+    public static final String PREFIX_PICTURE = "file://";
+    public static final String PICTURE_SAVE_LOCATION =
+            UserPrefs.FOLDER_LOCATION; // Where images are stored when added
+    public static final String DEFAULT_PICTURE_LOCATION =
+            "/images/";
+    public static final String DEFAULT_PICTURE =
+            "default_profile.png";
+    public static final String DEFAULT_ALEX =
+            "default_alex.png";
+    public static final String DEFAULT_BALAKRISHNAN =
+            "default_balakrishnan.png";
+    public static final String DEFAULT_BERNICE =
+            "default_bernice.png";
+    public static final String DEFAULT_CHARLOTTE =
+            "default_charlotte.png";
+    public static final String DEFAULT_DAVID =
+            "default_david.png";
+    public static final String DEFAULT_IRFAN =
+            "default_irfan.png";
+
+    private static final int RESIZE_IMAGE_WIDTH = 64;
+    private static final int RESIZE_IMAGE_HEIGHT = 64;
+
+    private static final String PICTURE_SUFFIX = ".png";
+    private static final String PICTURE_DELIMITER_SLASH = "/";
+    private static final String PICTURE_DELIMITER_BACKSLASH = "\\\\";
+    private static final int PICTURE_MAX_SIZE = 512000; // 512 KB
+
+    public final String value;
+
+    public Picture(String fileLocation) throws IllegalValueException {
+        String trimmedFileLocation = fileLocation == null ? null : fileLocation.trim();
+        if (!isValidPicture(trimmedFileLocation)) {
+            throw new IllegalValueException(MESSAGE_PROFILEPICTURE_CONSTRAINTS);
+        }
+
+        if (trimmedFileLocation != null) {
+            String[] split = trimmedFileLocation.split(PICTURE_DELIMITER_SLASH);
+            if (split.length < 2) {
+                split = trimmedFileLocation.split(PICTURE_DELIMITER_BACKSLASH);
+            }
+
+            // When we save the file, it is a single file name there is nothing to split.
+            // No need to copy it either
+
+            // last value before '/' is picture we want
+            String fileName = split[split.length - 1];
+
+            // length will give 1 when it is the file we saved
+            // in that case just put PICTURE_IMAGE_LOCATION to find it
+            if (split.length != 1) {
+                File src = new File(fileLocation);
+                File dest = new File(PICTURE_SAVE_LOCATION + fileName);
+
+                // If file is too big, resize it.
+                if (src.length() > PICTURE_MAX_SIZE) {
+                    resizeAndSaveImage(src, fileName);
+                } else {
+                    copyImage(src, dest);
+                }
+            }
+
+            this.value = fileName;
+        } else {
+            this.value = null;
+        }
+    }
```

Comment:
Method seems a bit too long. Try to SLAP more.
It is not a good sign when `else` appears far from the `if`